test/numbers.jl

```diff
@@ -2692,3 +2692,14 @@ end
         end
     end
 end
+
+@testset "constructor inferability for $T" for T in [AbstractFloat, #=BigFloat,=# Float16,
+        Float32, Float64, Integer, Bool, Signed, BigInt, Int128, Int16, Int32, Int64, Int8,
+        Unsigned, UInt128, UInt16, UInt32, UInt64, UInt8]
+    @test all(R -> R<:T, Base.return_types(T))
+end
+@testset "constructor inferability for BigFloat" begin
+    @test_broken all(R -> R<:T, Base.return_types(T))
```

Comment:
I think `T` is not defined here?

Reply:
D'Oh. I certainly did not intend the test to be _that_ broken. Fix incoming.